main.js

```diff
@@ -32,50 +35,62 @@ const printToDom = (divId, textToPrint) => {
   selectDiv.innerHTML = textToPrint;
 };
 
-const createForm = (domString) => {
-    `<form>
+const createForm = () => {
+  let form = `<form>
     <div class="mb-3">
     <label for="student" class="form-label">Student</label>
     <input type="text" class="form-control" id="student" required>
     </div>
-    <button type="submit" class="btn btn-primary" id="sort1">Sort!</button>
+    <button type="submit" class="btn btn-info" id="sort1">Sort!</button>
     </form>`
+    printToDom('#form', form);
+    document.querySelector('#sort1').addEventListener('click', cardCreated);
+    
   };
-  printToDom(domString);
 
   
   
-  function cardCreated(hatArr) {
-  let domString = '';
+  function cardCreated() {
+    let domString = '';
+    hat.forEach((element, i) => {
+      domString += `<div class="card mb-3" style="width: 18rem;" id=${i}>      
+        <div class="card-body">
+          <p class="card-text">${element.name}</p>
+          <p class="card-text">${element.values}</p>
+          <p class="card-text">${element.animal}</p>
+          <p class="card-text">${element.colors}</p>
+          <button type="button" class="btn btn-danger" id="${i}">Expel!</button>
+        </div>
+        </div>`
+    });
+    printToDom('#hats', domString);
+    
+  }
+
+
+const formInfo = (e) => {
```

Comment:
@cmeffley where is your students array? You need that for your data.

Reply:
I realized that I didn't have it this afternoon, I must have accidently deleted it. I added it while trying to get my Math.random to work. I also re-worked my hat array, to make things simpler.

Reply:
Awesome. Keep up the good work!